text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,1275 @@
+openapi: 3.0.1
+info:
+  title: IOTA REST API
+  description: 'This document describes the upcoming REST API for Hornet and Bee.'
+  contact:
+    email: contact@iota.org
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+externalDocs:
+  description: Find out more about IOTA
+  url: https://iota.org
+servers:
+  - url: https://127.0.0.1:8080/v1
+
+tags:
+  - name: node
+    description: Everything about the node itself.
+  - name: tangle
+    description: Everything about the tangle itself.
+  - name: messages
+    description: Everything about messages.
+  - name: transaction-messages
+    description: Everything about transaction-messages.
+paths:
+  /info:
+    get:
+      tags:
+        - node
+      summary: Returns information about the node.
+      description: Returns information about the node.
+      operationId: get-info
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GetInfoResponse'
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/get-info-response-example-1'
+  /tips:
+    get:
+      tags:
+        - tangle
+      summary: Returns two non-lazy tips.
+      description: Returns two non-lazy tips. In case the node can only provide one tip, tip1 and tip2 are identical.
+      operationId: get-tips
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GetTipsResponse'
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/get-tips-response-example-1'
+                Example 2:
+                  $ref: '#/components/examples/get-tips-response-example-2'
+  /messages/by-hash/{hash}:
+    get:
```

Comment:
I personally prefer to provide only either GET or POST version for each query APIs. Single query method isn't useful since multiple query can already do it imho. So how about we just leave a GET method for querying? `{hash}` can be a list of value. And since we probably want a maximum request boundry, it should be okay that GET URL length does has practical limit.

Reply:
Yes, will change that. Good idea.

Reply:
Changed. The desciptions of the endpoints are singular currently. Will try to come up with consistent plural forms; since there are some quite complex objects it's bit harder to come up with plural descriptions that makes sense.